flasc/visualization.py

```diff
@@ -637,14 +639,137 @@ def put_label(i):
         raise ValueError("Need one of near_i, near_x, near_y")
 
 
+def plot_binned_mean_and_ci(
+    x,
+    y,
+    color="b",
+    label="_nolegend_",
+    x_edges=None,
+    ax=None,
+    show_scatter=True,
+    show_bin_points=True,
+    show_confidence=True,
+    alpha_scatter=0.1,
+    confidence_level = 0.95,
+):
+    """
+    Plot data to a single axis.  Method
+    has options to include scatter of underlying data, specifiying
+    bin edges, and plotting confidence interval.
 
+    Args:
+        x (np.array): abscissa data.
+        y (np.array): ordinate data.
+        color (str, optional): line color.
+            Defaults to 'b'.
+        label (str, optional): line label used in legend.
+            Defaults to '_nolegend_'.
+        x_edges (np.array, optional): bin edges in x data
+            Defaults to None.
+        ax (:py:class:`matplotlib.pyplot.axes`, optional):
+            axes handle for plotting. Defaults to None.
+        show_scatter (bool, optional): flag to control scatter plot.
+            Defaults to True.
+        show_bin_points (bool, optional): flag to control plot of bins.
+            Defaults to True.
+        show_confidence (bool, optional): flag to control plot of
+            confidence interval. Defaults to True.
+        alpha_scatter (float, optional): Alpha for scatter
+            plot. Defaults to 0.5.
+        confidenceLevel (float, optional): Confidence level for
+            confidence interval. Defaults to 0.95.
 
+    """
 
+    # Check the length of x equals length of y
+    if len(x) != len(y):
+        raise ValueError("x and y must be the same length")
     
+    # Check that x is not empty
+    if len(x) == 0:
+        raise ValueError("x is empty")
     
 
-    
+    # Declare ax if not provided
+    if ax is None:
+        _, ax = plt.subplots()
 
+    # Put points ino dataframe
+    df = pd.DataFrame({"x": x, "y": y})
 
+    # If x_edges not provided, use 50 bins over range of x
+    if x_edges is None:
+        x_edges = np.linspace(df["x"].min()*.98, df["x"].max()*1.02, 50)
```

Comment:
I do see that the warnings that I was trying to avoid are still being raised in the example, I think because the data isn't really spread along the x axis in a realistic way. For now, I don't think that's really a problem.

Reply:
If you are using the example dataset, absolutely the data is mostly in really low wind speeds, indeed unrealistic. This has been an annoyance for me while working with this data, for sure. I wasn't sure if we wanted to fix this or just wait for the revision of the examples to finish.